src/main/java/com/develhope/spring/deals/services/OrderService.java

```diff
@@ -0,0 +1,58 @@
+package com.develhope.spring.deals.services;
+
+import com.develhope.spring.deals.models.Order;
+import com.develhope.spring.deals.models.OrderStatus;
+import com.develhope.spring.users.models.Roles;
+import com.develhope.spring.users.models.User;
+import com.develhope.spring.users.repositories.UserRepository;
+import com.develhope.spring.vehicles.models.Vehicle;
+import com.develhope.spring.vehicles.repositories.VehicleRepository;
+import com.develhope.spring.vehicles.vehicleEnums.MarketStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OrderService {
+
+    @Autowired
+    private VehicleRepository vehicleRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    public Order createOrderForBuyer(User buyer, long vehicleId) throws Exception {
+        if (buyer.getRoles() != Roles.BUYER) {
+            throw new Exception("User is not authorized to create an order as a buyer");
+        }
+
+        Vehicle vehicle = vehicleRepository.findById(vehicleId).orElseThrow(() -> new Exception("Vehicle not found"));
+
+        if (vehicle.getMarketStatus() != MarketStatus.ORDERABLE) {
+            throw new Exception("Vehicle is not orderable");
+        }
+
+        return new Order(true, OrderStatus.PENDING, (int) vehicle.getId());

```

Comment:
piuttosto cambiare long -> int qui, lo cambierei direttamente nella entity di Order in long